File: 21551-naveen-siva-kumar/week7/86-partition-list/86-partition-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* partition(ListNode* head, int x) {
        
        ListNode* sentinel = new ListNode(0,head);
        ListNode* fast = sentinel;
        ListNode* slow = sentinel;
        ListNode* temp;
        
        bool reachedX = false;
        while(fast->next!=NULL){// && fast->next-> < x){
            
            if(fast->next->val >= x){
                reachedX=true;
                fast = fast->next;
            }
            else if(fast->next->val < x && !reachedX){
                fast = fast->next;
                slow = fast;
            }
            else{
                temp = fast->next;
                fast->next = temp->next;
                temp->next = slow->next;
                slow->next = temp;
                slow = slow->next;
            }
        }
        return sentinel->next;
    }
};

File: 21557 - Mukesh J/Week-7/86-partition-list/86-partition-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode partition(ListNode head, int x) {
        ListNode leftdummy= new ListNode(0), rightdummy= new ListNode(0);
        ListNode slow=head, temp=rightdummy,temp1=leftdummy;
        while(slow!=null)
        {
            if(slow.val<x)
            {
                leftdummy.next=slow;
                leftdummy= leftdummy.next;
            }
            else
            {
                rightdummy.next=slow;
                rightdummy=rightdummy.next;
            }
            slow= slow.next;
        }
        leftdummy.next=temp.next;
        rightdummy.next=null;
        return temp1.next;
    }
}